Assets/Scripts/SS3D/Core/Systems/Rounds/RoundSystem.cs

```diff
@@ -0,0 +1,211 @@
+            InstanceFinder.ServerManager.Broadcast(warmupStartedMessage);
+        }
+
+        [Server]
+        private void HandleStartRound()
+        {
+            // Only do SyncVar assignments, tick coroutine and the RPC on the server.
+            if (!IsServer)
+            {
+                return;
+            }
+
+            if (RoundRunning)
+            {
+                Debug.Log($"[{nameof(RoundSystem)}] - Can't start round as round is already running");
+                return;
+            }
+            
+            UpdateRoundState(RoundState.Starting);
+            
+            UpdateRoundState(RoundState.Running);
+            StopCoroutine(_warmupCoroutine);
+            _tickCoroutine = StartCoroutine(Tick());
+            
+            InstanceFinder.ServerManager.Broadcast(new RoundStartedMessage());
+        }
+
+        [Server]
+        private IEnumerator TickWarmup()
+        {
+            while (_currentTimerSeconds > 0)
+            {
+                UpdateClock(GetTimerSeconds());
+                Debug.Log($"[{nameof(RoundSystem)}] - Start timer: {_currentTimerSeconds}");
+                _currentTimerSeconds--;
+                yield return new WaitForSeconds(1);
+            }
+
+            HandleStartRound();
+        }
+
+        [Server]
+        private IEnumerator Tick()
+        {
+            if (!IsServer)
+            {
+                yield break;
+            }
+            
+            while (RoundRunning)
+            {
+                UpdateClock(GetTimerSeconds());
+                _currentTimerSeconds++;
+                yield return new WaitForSeconds(1);
+            }
+
+            Debug.Log($"[{nameof(RoundSystem)}] - Coroutine running while round is not active");
+        }
+
+        [Server]
+        private void UpdateClock(int time)
+        {
+            if (!IsServer)
+            {
+                return;
+            }
+            
+            RoundTickUpdatedMessage roundTickUpdatedMessage = new(time);
+            InstanceFinder.ServerManager.Broadcast(roundTickUpdatedMessage);
+        }
+
+        private int GetTimerSeconds()
+        {
+            TimeSpan timeSpan = TimeSpan.FromSeconds(_currentTimerSeconds);
+            int timer = (int)timeSpan.TotalSeconds;
+            
+            return timer;
+        }
+
+        [Server]
+        private void UpdateRoundState(RoundState newState)
+        {
+            _roundState = newState;
+            
+            Debug.Log($"[{nameof(RoundSystem)}] - Round state updated: [{newState}]");
+            
+            RoundStateUpdatedMessage roundStateUpdatedMessage = new(newState);
+            InstanceFinder.ServerManager.Broadcast(roundStateUpdatedMessage);
+        }
+        
+        /// <summary>
+        /// Used by FishNet to sync the round state
+        /// </summary>
+        private void SetRoundState(RoundState oldState, RoundState newState, bool AsServer)
```

Comment:
This hook method seems to be unnecessary - the same effect is achieved with only the SyncVar. Even if this method is retained, you should not need to set _roundState in it, as that should already be set before the callback occurs (according to the [SyncVar documentation](https://fish-networking.gitbook.io/docs/manual/guides/synchronizing/syncvar)).

The same points apply to the hooks for SetCurrentTimerSeconds and SetWarmupTimer below.